helix-term/src/commands.rs

```diff
@@ -138,13 +125,21 @@ fn align_view(doc: &Document, view: &mut View, align: Align) {
 /// A command is composed of a static name, and a function that takes the current state plus a count,
 /// and does a side-effect on the state (usually by creating and applying a transaction).
 #[derive(Copy, Clone)]
-pub struct Command(&'static str, fn(cx: &mut Context));
+pub struct Command {
+    name: &'static str,
+    fun: fn(cx: &mut Context),
+    doc: &'static str,
+}
 
 macro_rules! commands {
-    ( $($name:ident),* ) => {
+    ( $($name:ident, $doc:literal),* ) => {
```

Comment:
Can you please make this take in a `#[doc($doc:literal)]` like how it was previously? Doing `///` might be better. Later I have an idea to extend that to large description without changing anything.

Reply:
Large description to show in the infobox ? Shouldn't it be kept as compact as possible ?

Reply:
Large description won't be shown in infobox. But you can just put a short description like I did for now. If later we want to extend it with large description I have an idea, like how I did with the infobox title.